CLAUDE.md

```diff
@@ -0,0 +1,364 @@
+- `baseTextStyle`, `value`, `theme`, `location`
+- Use `Screens.CHANNEL` or appropriate constant from `@constants` for the `location` prop
+- `textStyles`, `blockStyles`, `enableLatex`, and similar props are auto-generated via HOC
+
+**Important Limitations:**
+- **Cannot embed custom React components inline**: The Markdown component renders CommonMark to native components. You cannot insert custom React components (like badges, icons) inline within the markdown text flow
+- **Link handling**: Links in markdown go through `openLink()` → deep link system → `tryOpenURL()`. Custom URL schemes (e.g., `citation:`, `action:`) will show error alerts unless you implement a full deep link handler in `app/utils/deep_link/`
+- **Post-processing not possible**: Unlike web, you cannot post-process the rendered markdown JSX tree to replace markers with React components
+
+### Adding New Post Types
+When adding custom post types (e.g., for new products):
+1. Add the type string to `PostType` union in `types/api/posts.d.ts`
+2. Define constants in your product's constants file (e.g., `app/products/agents/constants.ts`)
+3. Use the globally-available `Post` type (defined in `types/api/posts.d.ts`) - no import needed
+
+### Platform-Specific
+- iOS: Opt out of iOS 18+ features (liquid glass) until UI is properly addressed
+- Android: Test on multiple API levels (34, 35, 36) to verify behavior with edge-to-edge changes
+- Use `Set` instead of `Array` for exception lists (faster, ensures uniqueness)
+
+### testID Convention
+Components use hierarchical testIDs: `component.subcomponent.element`
+- Example: `channel_list.category.CATEGORY.channel_item.ID.display_name`
+- Example: `channel.post_draft.send_action.send.button`
+
+## Common Mistakes to Avoid
+
+### JavaScript Compatibility
+- **Don't use `Object.hasOwn()`** - React Native doesn't support ES2022+ features
+- ❌ `Object.hasOwn(obj, 'key')`
+- ✅ `'key' in obj`
+
+### Anti-Patterns
+- Conditionals in tests for TypeScript satisfaction
+- Creating new arrays/objects on every render when passed as props
+- Over-engineering - follow YAGNI principle
+- Impossible states with boolean + optional parameters that depend on each other
+- Redundant checks - if `array.length >= 3`, indices 0,1,2 are guaranteed to exist
+- Verbose error messages that reveal internal system details
+- Mass assignment risks - whitelist fields instead of sending `Partial<Model>` objects
+- Duplicate conditional branches that return identical results - consolidate into a single condition (e.g., `if (A && B)` and `if (A && !B)` both returning same thing should be just `if (A)`)
+
+### Security
+- Question whether showing secrets with "eye" toggle is a security concern
+- Don't commit files that likely contain secrets (.env, credentials.json)
+- Consider pre-populating sensitive fields with `*********` instead of actual values
+
+## Configuration
+
+- **Base config**: `dist/assets/config.json` (feature flags, white-labeling)
```

Comment:
`assets/base/config.json` -> dist is generated by running `node ./scripts/generate-assets.js`